shared/pkg/formatter/syft.go

```diff
@@ -60,29 +60,26 @@ func (f *Syft) GetSBOMBytes() []byte {
 }
 
 func (f *Syft) Encode(format string) error {
-	syftFormat := getSyftFormat(format)
+	// TODO maybe check cyclonedx format as output
+	sbomFormat := syft.FormatByName(format)
 	var err error
-	f.sbomBytes, err = syft.Encode(f.sbomStruct, syftFormat)
+	f.sbomBytes, err = syft.Encode(f.sbomStruct, sbomFormat)
 	if err != nil {
 		return fmt.Errorf("failed to encode SBOM: %v", err)
 	}
 	return nil
 }
 
 func (f *Syft) Decode(format string) error {
-	return nil
-}
+	reader := bytes.NewReader(f.sbomBytes)
```

Comment:
maybe worth checking the input format?

Reply:
syft.Decode(reader) will identify the input format and return an error if it's failed.